src/main/java/com/descope/proxy/impl/AbstractProxyImpl.java

```diff
@@ -137,9 +137,16 @@ private static <R> Supplier<R> toSupplierOfType(
           ByteArrayOutputStream bs = new ByteArrayOutputStream();
           TeeInputStream tee = new TeeInputStream(stream, bs, true);
           if (responseInfo.statusCode() < 200 || responseInfo.statusCode() > 299) {
+            if (responseInfo.statusCode() == 429) { // Rate limit from infra
+              throw new RateLimitExceededException(
+                "Rate limit exceeded",
+                ErrorCode.RATE_LIMIT_EXCEEDED,
+                responseInfo.headers().firstValueAsLong(RETRY_AFTER_HEADER).orElse(DEFAULT_RETRY));              
+            }
             try {
               var errorDetails = objectMapper.readValue(tee, JsonBodyHandler.ErrorDetails.class);
               log.error(errorDetails.getActualMessage());
+              log.error(bs.toString());
```

Comment:
this change is deliberately? of debugging leftover?

Reply:
Deliberate - because errors sometimes have weird fields and it's good to display the entire body

Reply:
Dont we want to add some hard coded text to our logs, so we will be able to tell from where they came ?